bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -143,72 +144,102 @@ pub fn results_sonobuoy(
         serde_json::from_str(&stdout).context(error::DeserializeJsonSnafu)?;
     trace!("The sonobuoy results are valid json");
 
-    let e2e_status = run_status
-        .get("plugins")
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .as_array()
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .first()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}", sonobuoy_config.plugin),
-        })?;
-
-    // Sometimes a helpful log is available in the progress field, but not always.
-    let progress_status = e2e_status
-        .get("progress")
-        .map(|value| value.to_string())
-        .unwrap_or_else(|| "".to_string());
-
-    let result_status = e2e_status
-        .get("result-status")
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?
-        .as_str()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?;
-
-    let result_counts = run_status
+    process_sonobuoy_test_results(&run_status)
+}
+
+/// process_sonobuoy_test_results parses the output from `sonobuoy status --json` output and gets
+/// the overall status of the plugin results.
+pub(crate) fn process_sonobuoy_test_results(
+    run_status: &serde_json::Value,
+) -> Result<TestResults, error::Error> {
+    let mut num_passed: u64 = 0;
+    let mut num_failed: u64 = 0;
+    let mut num_skipped: u64 = 0;
+    let mut progress = Vec::new();
+    let mut outcome_summary = HashMap::from([
+        ("pass", 0),
+        ("passed", 0),
+        ("fail", 0),
+        ("failed", 0),
+        ("timeout", 0),
+        ("timed-out", 0),
+    ]);
+
+    let plugin_results = run_status
         .get("plugins")
         .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
         .as_array()
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .first()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}", sonobuoy_config.plugin),
-        })?
-        .get("result-counts")
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-counts", sonobuoy_config.plugin),
-        })?;
-
-    let num_passed = result_counts
-        .get("passed")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
-
-    let num_failed = result_counts
-        .get("failed")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
-
-    let num_skipped = result_counts
-        .get("skipped")
-        .map(|v| v.as_u64().unwrap_or(0))
-        .unwrap_or(0);
+        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?;
+
+    for result in plugin_results {
+        let plugin = result
```

Comment:
Do we have somewhere in `TestResults` that we could report the list of plugins that reported failuers? I guess we could put in OtherInfo along with the progress report(s)?

Reply:
So maybe something like:

```rust
if (results_status == "fail" || results_status == "failed") && progress_status.is_empty() {
    progress.push(format!("{}: Failed", plugin);
}
```